pkg/ctl/cmdutils/nodegroup_filter_test.go

```diff
@@ -14,9 +14,122 @@ import (
 
 var _ = Describe("nodegroup filter", func() {
 
-	Context("CheckEachNodeGroup", func() {
+	Context("Match", func() {
+		var (
+			filter *NodeGroupFilter
+			cfg    *api.ClusterConfig
+		)
+
+		BeforeEach(func() {
+			cfg = newClusterConfig()
+			addGroupA(cfg)
+			addGroupB(cfg)
+
+			filter = NewNodeGroupFilter()
+		})
+
+		It("should match empty filter", func() {
+			included, excluded := filter.MatchAll(cfg.NodeGroups)
+			Expect(included).To(HaveLen(6))
+			Expect(excluded).To(HaveLen(0))
+			Expect(included.HasAll("test-ng1a", "test-ng2a", "test-ng3a", "test-ng1b", "test-ng2b", "test-ng3b")).To(BeTrue())
+		})
+
+		It("should match exclude filter with ExcludeAll", func() {
+			filter.ExcludeAll = true
+			included, excluded := filter.MatchAll(cfg.NodeGroups)
+			Expect(included).To(HaveLen(0))
+			Expect(excluded).To(HaveLen(6))
+			Expect(excluded.HasAll("test-ng1a", "test-ng2a", "test-ng3a", "test-ng1b", "test-ng2b", "test-ng3b")).To(BeTrue())
+		})
+
+		It("should match exclude filter with names and globs", func() {
+			filter.ExcludeNames.Insert("test-ng3b")
```

Comment:
now that I see how this is used, maybe this insert should be exposed as a method too like `filter.AddExcludedName(name)`. That way we don't leak an impl detail (the set, which later could be an array or it could be made thread safe if needed by adding a mutex, etc)

Reply:
Yeah, I was just lazy, thanks.